### PR TITLE
PHOENIX-7417 Remove commons-collections dependency

### PR DESCRIPTION
**Pull request by NihalJain**
- Remove imports and replace with equivalent code
- Add RestrictImports rule to restrict further usage of it in codebase

### CODE REVIEW

**Comment by NihalJain** [comment]
We have same in query server also. Do we use same ticket for both PRs? Or raise new Jira for queryserver PR?

**Comment by NihalJain** [comment]
Saw an import issue in code change. Went ahead and raised https://github.com/apache/phoenix/pull/1995/files as fixing manually is too cumbersome

**Comment by stoty** [comment]
There is minimal functionality that we use here.

Wouldn't it be better to just get rid of this dependency ?

**Comment by NihalJain** [comment]
> Wouldn't it be better to just get rid of this dependency ?

Sure let me see if I can remove the usage of commons-collections altogether

**Comment by NihalJain** [comment]
> > Wouldn't it be better to just get rid of this dependency ?
> 
> Sure let me see if I can remove the usage of commons-collections altogether

Done!

File: phoenix-core-client/src/main/java/org/apache/phoenix/util/InstanceResolver.java [hunk 1]
```diff
@@ -71,14 +70,22 @@ public static <T> T getSingleton(Class<T> clazz, T defaultInstance) {
...
+            defaultInstances.addAll(iteratorToList(iterator));
         } else {
-            defaultInstances = IteratorUtils.toList(iterator);
+            defaultInstances = iteratorToList(iterator);
         }
 
         return defaultInstances;
     }
 
+    private static <T> List<T> iteratorToList(Iterator<T> iterator) {
```
Comment (stoty) [inline phoenix-core-client/src/main/java/org/apache/phoenix/util/InstanceResolver.java:81]: nit: While there is nothing wrong with this, here are a few solutions that do not need a new method:

https://stackoverflow.com/questions/10117026/convert-iterator-to-list

Comment (stoty) [reply phoenix-core-client/src/main/java/org/apache/phoenix/util/InstanceResolver.java:81]: I mean the guava or lambd based ones.

**Review by stoty — APPROVED** [review]
+1 LGTM

You may want to address my nit, but it's fine either way.

File: phoenix-core-client/src/main/java/org/apache/phoenix/util/InstanceResolver.java (hunk 1 above)
Comment (NihalJain) [reply phoenix-core-client/src/main/java/org/apache/phoenix/util/InstanceResolver.java]: done

**Review by stoty — APPROVED** [review]
+1 LGTM

**Comment by stoty** [comment]
please fix the checkstyle errors before commit (unless they are pre-existing)

**Comment by NihalJain** [comment]
> please fix the checkstyle errors before commit (unless they are pre-existing)

/phoenix-core-client/src/main/java/org/apache/phoenix/util/QueryBuilder.java:150:    is new let me fix this

**Comment by stoty** [comment]
Still +1

**Comment by NihalJain** [comment]
Thank you @stoty I plan to fix this for other phoenix projects. PQS also depends on commons-collections. Raised PHOENIX-7430.

**Pull request merged**